core/domain/LDAStringClassifierTest.py

```diff
@@ -231,3 +231,62 @@ def test_predict_label_for_doc(self):
         labels = self.classifier.predict(self._EXAMPLES_TEST)
         self.assertEquals(labels[2], '_default')
         self._validate_instance()
+
+    def test_validate(self):
+        """Test to verify validate method of LDAStringClassifier."""
+
+        # Verify no errors are raised for correct data.
+        classifier_data = {
+            '_alpha': 0.1,
+            '_beta': 0.001,
+            '_prediction_threshold': 0.5,
+            '_training_iterations': 25,
+            '_prediction_iterations': 5,
+            '_num_labels': 10,
+            '_num_docs': 12,
+            '_num_words': 20,
+            '_label_to_id': {'text': 1},
+            '_word_to_id': {'hello': 2},
+            '_w_dp': [],
+            '_b_dl': [],
+            '_l_dp': [],
+            '_c_dl': [],
+            '_c_lw': [],
+            '_c_l': []
+        }
+        self.classifier.validate(classifier_data)
+
+        # Verify Type error is raised when string is provided instead of int.
+        classifier_data['_alpha'] = 'abc'
+        with self.assertRaises(TypeError):
```

Comment:
For all these tests, could you do self.assertRaisesRegexp() and check against some substring of the error message that is specific to the given error? Otherwise it's hard to tell whether the correct error case is being triggered.